packages/eslint-plugin/src/rules/no-shadow.ts

```diff
@@ -22,6 +22,8 @@ const allowedFunctionVariableDefTypes = new Set([
   AST_NODE_TYPES.TSCallSignatureDeclaration,
   AST_NODE_TYPES.TSFunctionType,
   AST_NODE_TYPES.TSMethodSignature,
+  AST_NODE_TYPES.TSEmptyBodyFunctionExpression,
+  AST_NODE_TYPES.TSDeclareFunction,
```

Comment:
Hi @ronami IMO, `TSConstructSignatureDeclaration` should also be added, what do you think?

- [playground](https://deploy-preview-10543--typescript-eslint.netlify.app/play/#ts=5.7.2&fileType=.tsx&code=MYewdgzgLgBAhgJwOYwLwwAwG4BQOAmApsADaKEyiSwBiIIAXDAN44wwBm9TAFIkkzABXALYAjQggCUaAHwwAbiACW%2BXOzCEA7n2SDRE6UyWqsMAPTmYAcn7WYyiPBIJCcfAE8YRUuXwOwGCgACwohAAdwyRgIUCiYcBgSZU0YACZKEBJRQIB2ADocAF8sIA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6Jge1uQAsBDACYcA7ujABtSImjQO0SAF1wYAL4gVQA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false)
```ts
// "ignoreFunctionTypeParameterNameValueShadow": true
const arg = 0;

declare const Foo: {
  foo: (arg: number) => void;
  new(arg: number): void; // 'arg' is already declared in the upper scope on line 2 column 7.
};
```

Reply:
Thank you @yeonjuan for catching this 👍

Reply:
@ronami I edited the previous comment :) . I think we should add `TSConstructorType` also

Reply:
Oh, I just went over the list of nodes to make sure I didn't miss anything 🙈 

Thanks!

Reply:
LGTM! Thanks! 👍

Reply:
Thanks @yeonjuan for the catch!